src/test/java/com/google/step/youtube/YoutubeInfoScraperTest.java

```diff
@@ -115,6 +118,40 @@ public void scrapeChannelUploadPlaylist_IOException() throws IOException {
         assertThrows(IOException.class, () -> scraper.scrapeChannelUploadPlaylist(IOEXCEPTION));
     }
 
+    @Test
+    public void scrapeUserUploadPlaylist_nonExistentUserName()  throws IOException {
+        mockChannelResponse = new ChannelListResponse();
+        when(mockListChannels.execute()).thenReturn(mockChannelResponse.setItems(null));
+        Optional<String> actual = scraper.scrapeUserUploadPlaylist(NONEXISTENT_USER_NAME);
+        assertThat(actual.isPresent(), equalTo(false));
+    }
+
+    @Test
+    public void scrapeUserUploadPlaylist_emptyList()  throws IOException {
+        mockChannelResponse = new ChannelListResponse();
+        when(mockListChannels.execute()).thenReturn(mockChannelResponse.setItems(Arrays.asList()));
+        Optional<String> actual = scraper.scrapeUserUploadPlaylist(USER_NAME);
+        assertThat(actual.isPresent(), equalTo(false));
+    }
+
+    @Test
+    public void scrapeUserUploadPlaylist_userExists() throws IOException {
+        mockChannelResponse = new ChannelListResponse();
+        Channel channel = new Channel();
+        channel.setContentDetails(
+                new ChannelContentDetails().setRelatedPlaylists(new RelatedPlaylists().setUploads(UPLOAD_ID)));
+        mockChannelResponse.setItems(Arrays.asList(channel));
+        when(mockListChannels.execute()).thenReturn(mockChannelResponse);
+        Optional<String> actual = scraper.scrapeUserUploadPlaylist(USER_NAME);
+        assertThat(actual.get(), equalTo(UPLOAD_ID));
+    }
+
+    @Test
+    public void scrapeUserUploadPlaylist_IOException() throws IOException {
+        when(mockListChannels.execute()).thenThrow(IOException.class);
+        assertThrows(IOException.class, () -> scraper.scrapeUserUploadPlaylist(IOEXCEPTION));
```

Comment:
the constant IOEXCEPTION isn't actually an exception.  it's actually a user...  Also I think any string that you pass will still result in an io exception.  you might as well just pass USER_NAME